src/aotcompile.cpp

```diff
@@ -344,6 +358,47 @@ void *jl_create_native_impl(jl_array_t *methods, LLVMOrcThreadSafeModuleRef llvm
     }
     CreateNativeMethods += emitted.size();
 
+    size_t offset = gvars.size();
+    data->jl_external_to_llvm.resize(params.external_fns.size());
+
+    for (auto &extern_fn : params.external_fns) {
+        jl_code_instance_t *this_code = std::get<0>(extern_fn.first);
+        bool specsig = std::get<1>(extern_fn.first);
+        assert(specsig && "Error external_fns doesn't handle non-specsig yet");
+        (void)specsig;
+        Function *F = extern_fn.second;
+        Module *M = F->getParent();
+
+        Type *T_funcp = F->getFunctionType()->getPointerTo();
+        // Can't create a GC with type FunctionType. Alias also doesn't work
+        GlobalVariable *GV = new GlobalVariable(*M, T_funcp, false,
+                                                GlobalVariable::ExternalLinkage,
+                                                Constant::getNullValue(T_funcp),
+                                                F->getName());
+
+
+        // Need to insert load instruction, thus we can't use replace all uses with
+        for (Value *Use: F->users()) {
+            if (auto CI = dyn_cast<CallInst>(Use)) {
+                auto Callee = new LoadInst(T_funcp, GV, "", false, Align(1), CI); // TODO correct Align?
+                CI->setCalledOperand(Callee);
+                continue;
+            } else {
+                llvm::outs() << *Use << "\n";
+                assert(false);
+            }
+        }
```

Comment:
Multiversioning has the right way of dealing with this RAUW, we might want to export it from there